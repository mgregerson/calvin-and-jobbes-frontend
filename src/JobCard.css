.JobCard {
  border: none;
  border-radius: 0;
  margin-bottom: 1rem;
  max-height: 250px;
  max-width: 800px;
  text-decoration: none;
  display: flex;
}

.card-title {
  justify-content: space-between;
}

.card {
  display: flex;
  flex-direction: column;
  min-width: 0;
  position: relative;
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
  color: inherit;
}
