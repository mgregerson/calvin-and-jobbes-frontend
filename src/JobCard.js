import "./JobCard.css";

/** JobCard
 *
 * Props:
 *      - title
 *      - companyName
 *      - salary
 *      - equity
 *
 * Renders individual Job Card
 *
 * JobCardList -> JobCard
 */

function JobCard({ title, companyName, salary, equity }) {
  return (
    <div className="JobCard card my-2">
      <div className="JobCard-body">
        <h6 className="JobCard-title">{title}</h6>
        <p className="JobCard-company-name">{companyName}</p>
        {salary !== null ? (
          <div>
            <small className="JobCard-salary">Salary: {salary}</small>
          </div>
        ) : (
          <div>
            <small className="JobCard-salary">Unpaid Internship</small>
          </div>
        )}
        {equity !== null && (
          <div>
            <small className="JobCard-equity">Equity: {equity}</small>
          </div>
        )}
      </div>
    </div>
  );
}

export default JobCard;
