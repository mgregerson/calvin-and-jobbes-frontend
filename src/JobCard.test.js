import JobCard from "./JobCard";
import { render } from "@testing-library/react";

const TEST_JOB = {
  title: "Test Job",
  salary: 100000,
  equity: 0.1,
  companyName: "Test Company",
};

test("renders without crashing", function () {
  render(
    <JobCard
      title={TEST_JOB.title}
      salary={TEST_JOB.salary}
      equity={TEST_JOB.equity}
      companyName={TEST_JOB.companyName}
    />
  );
});

test("matches snapshot", function () {
  const { container } = render(
    <JobCard
      title={TEST_JOB.title}
      salary={TEST_JOB.salary}
      equity={TEST_JOB.equity}
      companyName={TEST_JOB.companyName}
    />
  );
  expect(container).toMatchSnapshot();
});

test("Displays all information on the job on the page if salary/equity NOT null", () => {
  const { container } = render(
    <JobCard
      title={TEST_JOB.title}
      salary={TEST_JOB.salary}
      equity={TEST_JOB.equity}
      companyName={TEST_JOB.companyName}
    />
  );

  expect(container.querySelector(".JobCard-title").innerHTML).toEqual(
    "Test Job"
  );
  expect(container.querySelector(".JobCard-company-name").innerHTML).toEqual(
    "Test Company"
  );
  expect(container.querySelector(".JobCard-salary").innerHTML).toEqual(
    "Salary: 100000"
  );
  expect(container.querySelector(".JobCard-equity").innerHTML).toEqual(
    "Equity: 0.1"
  );
});

test("Displays as unpaid internship if salary is null", () => {
  const { container } = render(
    <JobCard
      title={TEST_JOB.title}
      salary={null}
      equity={TEST_JOB.equity}
      companyName={TEST_JOB.companyName}
    />
  );

  expect(container.querySelector(".JobCard-title").innerHTML).toEqual(
    "Test Job"
  );
  expect(container.querySelector(".JobCard-company-name").innerHTML).toEqual(
    "Test Company"
  );
  expect(container.querySelector(".JobCard-salary").innerHTML).toEqual(
    "Unpaid Internship"
  );
  expect(container.querySelector(".JobCard-equity").innerHTML).toEqual(
    "Equity: 0.1"
  );
});

test("Does not display equity if null", () => {
  const { container } = render(
    <JobCard
      title={TEST_JOB.title}
      salary={TEST_JOB.salary}
      equity={null}
      companyName={TEST_JOB.companyName}
    />
  );

  expect(container.querySelector(".JobCard-title").innerHTML).toEqual(
    "Test Job"
  );
  expect(container.querySelector(".JobCard-company-name").innerHTML).toEqual(
    "Test Company"
  );
  expect(container.querySelector(".JobCard-salary").innerHTML).toEqual(
    "Salary: 100000"
  );
  expect(container.querySelector(".JobCard-equity")).not.toBeInTheDocument();
});
