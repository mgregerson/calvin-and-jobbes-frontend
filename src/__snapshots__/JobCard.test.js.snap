// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="JobCard card my-2"
  >
    <div
      class="JobCard-body"
    >
      <h6
        class="JobCard-title"
      >
        Test Job
      </h6>
      <p
        class="JobCard-company-name"
      >
        Test Company
      </p>
      <div>
        <small
          class="JobCard-salary"
        >
          Salary: 
          100000
        </small>
      </div>
      <div>
        <small
          class="JobCard-equity"
        >
          Equity: 
          0.1
        </small>
      </div>
    </div>
  </div>
</div>
`;
